Sem. 03/Pract. 03/mySolutions/Task02.cpp

```diff
@@ -0,0 +1,329 @@
+#include <iostream>
+#include <fstream>
+
+const char* ALL_OFFERS_BINARY_FILE_PATH = "all_offers.bin";
+const char* PERFECT_OFFERS_BINARY_FILE_PATH = "perfect_offers.bin";
+const size_t MAX_COMPANY_NAME_LENGTH = 26;
+
+struct JobOffer
+{
+	char companyName[MAX_COMPANY_NAME_LENGTH];
+	int coworkers;
+	int numberOfVacancyDays;
+	long long salary;
+};
+
+size_t strLength(const char* str)
+{
+	size_t length = 0;
+
+	while (*str != 0)
+	{
+		str++;
+		length++;
+	}
+
+	return length;
+}
+
+int strCompare(const char* str1, const char* str2)
+{
+	size_t str1Len = strLength(str1);
+	size_t str2Len = strLength(str2);
+
+	size_t lenToUse = str1Len >= str2Len ? str1Len : str2Len;
+
+	for (size_t i = 0; i < lenToUse; i++)
+	{
+		if (str1[i] > str2[i])
+			return -1;
+
+		if (str1[i] < str2[i])
+			return 1;
+	}
+
+	return 0;
+}
+
+size_t getFileSize(std::ifstream& in)
+{
+	size_t currentPos = in.tellg();
+	in.seekg(0, std::ios::end);
+	size_t fileSize = in.tellg();
+	in.seekg(currentPos);
+	return fileSize;
+}
+
+
+void strCopy(char* direction, const char* toCopy)
+{
+	size_t toCopyLen = strLength(toCopy);
+
+	for (size_t i = 0; i < toCopyLen; i++)
+		direction[i] = toCopy[i];
+
+	direction[toCopyLen] = '\0';
+}
+
+void saveJobOffer(std::ofstream& out, const JobOffer& offer)
+{
+	out.write((const char*)&offer, sizeof(JobOffer));
+}
+
+JobOffer readJobOfferFromFile(std::ifstream& in)
+{
+	JobOffer offer;
+	in.read((char*)&offer, sizeof(JobOffer));
+	return offer;
+}
+
+bool existOffer(const char* filePath, const char* companyName)
+{
+	std::ifstream in(filePath, std::ios::binary);
+
+	if (!in.is_open())
+	{
+		std::cerr << "Couldn't find file!";
+		return false;
+	}
+
+	while (!in.eof())
+	{
+		JobOffer offer = readJobOfferFromFile(in);
+
+		if (strCompare(offer.companyName, companyName) == 0)
+			return true;
+	}
+
+	in.close();
+	return false;
```

Comment:
it's faster if you read all offers at once and then check them. The reading from file is slow operation